armory/data/datasets.py

```diff
@@ -678,19 +679,52 @@ def _cache_dataset(dataset_dir: str, name: str, subpath: str):
 }
 
 
-def download_all():
+def download_all(dataset_names):
     """
-    Download all datasets to cache.
+    Download all datasets for a scenario or requested datset to cache.
     """
-    errors = []
-    for name, func in SUPPORTED_DATASETS.items():
-        logger.info(f"Downloading (if necessary) dataset {name}")
-        try:
-            func()
-        except Exception:
-            errors.append(name)
-            logger.exception(f"Loading dataset {name} failed.")
-    if errors:
-        logger.info("All datasets downloaded successfully")
+    if dataset_names.lower().endswith(".json"):
+        datasets_to_download = _read_validate_data_download_config(dataset_names)[
+            "datasets"
+        ]["names"]
+        for dataset_to_download in datasets_to_download:
+            _download_data(dataset_to_download)
+    else:
+        _download_data(dataset_names)
+
+
+def _download_data(dataset_name):
+    """
+    Download a single dataset to cache.
+    """
+    if dataset_name not in SUPPORTED_DATASETS.keys():
+        raise ValueError(
+            f"dataset {dataset_name} not supported. Must be one of {list(SUPPORTED_DATASETS.keys())}"
+        )
+
+    func = SUPPORTED_DATASETS[dataset_name]
+
+    logger.info(f"Downloading (if necessary) dataset {dataset_name}...")
+
+    if dataset_name == "imagenet_adversarial":
+        kwargs = {"epochs": 1, "batch_size": 1}
```

Comment:
If there were kwarg defaults for all of the arguments for the dataset functions (split_type="train" or "clean", epochs=1, batch_size=1), then we wouldn't need to specify here - we could just call the dataset function to download.